### PR TITLE
Drop-In UI : Landscape designs implementation

### PR DESCRIPTION
**Pull request by abhishek1508**
### Description

Fixes https://github.com/mapbox/navigation-sdks/issues/1717

### Screenshots or Gifs

https://user-images.githubusercontent.com/9770186/168907078-68180666-7c7d-4f5f-9e11-724e4f18f902.mp4

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ This release depends on, and has been tested with, the following Mapbox dependen
 ## Mapbox Navigation SDK 2.5.0-rc.2 - May 19, 2022
 ### Changelog
 [Changes between v2.5.0-rc.1 and v2.5.0-rc.2](https://github.com/mapbox/mapbox-navigation-android/compare/v2.5.0-rc.1...v2.5.0-rc.2)
+- Added capabilities for Drop-In UI to render in landscape mode. [#5823](https://github.com/mapbox/mapbox-navigation-android/pull/5823)
```
Comment (kmadsen) [inline CHANGELOG.md:27]: Move this up to the unreleased

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/camera/CameraLayoutObserver.kt [hunk 2]
```diff
@@ -37,18 +53,45 @@ internal class CameraLayoutObserver(
...
+        val left = mapView.width.toDouble() - binding.infoPanelLayout.right.toDouble()
+        val right = mapView.width.toDouble() - binding.actionListLayout.left.toDouble()
+        val bottom = mapView.height.toDouble() - binding.roadNameLayout.top.toDouble()
+        return when (store.state.value.navigation) {
+            is NavigationState.FreeDrive -> {
+                EdgeInsets(
+                    vPaddingLandscape,
+                    hPaddingLandscape,
+                    vPaddingLandscape.plus(bottom),
+                    hPaddingLandscape
```
Comment (kmadsen) [inline libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/camera/CameraLayoutObserver.kt:82]: Curious if `left` `right` and `bottom` are all zero here. They're already calculated so this block may be unnecessary

**Comment by kmadsen** [comment]
It may make sense to put the sliders on the left/right side for the customized activity

Screen Shot 2022-05-19 at 4 08 15 PM

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/camera/CameraLayoutObserver.kt (hunk 2 above)
Comment (abhishek1508) [reply libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/camera/CameraLayoutObserver.kt:82]: > left right and bottom

I guess you mean left, right and top

They are not 0 as if you see `vPaddingLandscape` and `hPaddingLandscape` have values. These are the padding values that needs to be applied to the camera. The bottom applies the padding + height of road name label

**Comment by kmadsen** [comment]
nit: considering this unrelated to this change (because changing orientation surfaces memory leaks). But here's a log while testing the "Navigation View in a Fragment test"

```
2022-05-19 16:13:57.534 11947-11947/com.mapbox.navigation.qa_test_app E/Mapbox: [maps-android\Mbgl-MapboxMap] Mapbox SDK memory leak detected! MapboxMap object (accessing setCamera) should not be stored and used after MapView is destroyed.
2022-05-19 16:13:57.534 11947-11947/com.mapbox.navigation.qa_test_app E/Mapbox: [maps-android\Mbgl-MapboxMap] Mapbox SDK memory leak detected! MapboxMap object (accessing cameraState) should not be stored and used after MapView is destroyed.
2022-05-19 16:13:57.546 11947-11947/com.mapbox.navigation.qa_test_app E/Mapbox: [maps-android\Mbgl-MapboxMap] Mapbox SDK memory leak detected! MapboxMap object (accessing setCamera) should not be stored and used after MapView is destroyed.
2022-05-19 16:13:57.546 11947-11947/com.mapbox.navigation.qa_test_app E/Mapbox: [maps-android\Mbgl-MapboxMap] Mapbox SDK memory leak detected! MapboxMap object (accessing cameraState) should not be stored and used after MapView is destroyed.
2022-05-19 16:13:57.557 11947-11947/com.mapbox.navigation.qa_test_app E/Mapbox: [maps-android\Mbgl-MapboxMap] Mapbox SDK memory leak detected! MapboxMap object (accessing setCamera) should not be stored and used after MapView is destroyed.
2022-05-19 16:13:57.557 11947-11947/com.mapbox.navigation.qa_test_app E/Mapbox: [maps-android\Mbgl-MapboxMap] Mapbox SDK memory leak detected! MapboxMap object (accessing cameraState) should not be stored and used after MapView is destroyed.
2022-05-19 16:13:57.568 11947-11947/com.mapbox.navigation.qa_test_app E/Mapbox: [maps-android\Mbgl-MapboxMap] Mapbox SDK memory leak detected! MapboxMap object (accessing setCamera) should not be stored and used after MapView is destroyed.
2022-05-19 16:13:57.568 11947-11947/com.mapbox.navigation.qa_test_app E/Mapbox: [maps-android\Mbgl-MapboxMap] Mapbox SDK memory leak detected! MapboxMap object (accessing cameraState) should not be stored and used after MapView is destroyed.
2022-05-19 16:13:57.579 11947-11947/com.mapbox.navigation.qa_test_app E/Mapbox: [maps-android\Mbgl-MapboxMap] Mapbox SDK memory leak detected! MapboxMap object (accessing setCamera) should not be stored and used after MapView is destroyed.
2022-05-19 16:13:57.579 11947-11947/com.mapbox.navigation.qa_test_app E/Mapbox: [maps-android\Mbgl-MapboxMap] Mapbox SDK memory leak detected! MapboxMap object (accessing cameraState) should not be stored and used after MapView is destroyed.
2022-05-19 16:13:57.589 11947-11947/com.mapbox.navigation.qa_test_app E/Mapbox: [maps-android\Mbgl-MapboxMap] Mapbox SDK memory leak detected! MapboxMap object (accessing setUserAnimationInProgress) should not be stored and used after MapView is destroyed.
2022-05-19 16:13:57.590 11947-11947/com.mapbox.navigation.qa_test_app E/Mapbox: [maps-android\Mbgl-MapboxMap] Mapbox SDK memory leak detected! MapboxMap object (accessing setCamera) should not be stored and used after MapView is destroyed.
2022-05-19 16:13:57.590 11947-11947/com.mapbox.navigation.qa_test_app E/Mapbox: [maps-android\Mbgl-MapboxMap] Mapbox SDK memory leak detected! MapboxMap object (accessing cameraState) should not be stored and used after MapView is destroyed.
2022-05-19 16:14:03.257 11947-11947/com.mapbox.navigation.qa_test_app D/LeakCanary: Watching instance of androidx.lifecycle.ReportFragment (androidx.lifecycle.ReportFragment received Fragment#onDestroy() callback) with key 3d89f38f-7ff9-4bd9-a725-c281d612b029
2022-05-19 16:14:03.258 11947-11947/com.mapbox.navigation.qa_test_app D/LeakCanary: Watching instance of com.mapbox.navigation.qa_test_app.view.MapboxNavigationViewFragmentActivity (com.mapbox.navigation.qa_test_app.view.MapboxNavigationViewFragmentActivity received Activity#onDestroy() callback) with key b78703de-9021-4292-8515-9cfe2922b7ad
2022-05-19 16:14:03.304 11947-11947/com.mapbox.navigation.qa_test_app D/LeakCanary: Watching instance of com.mapbox.navigation.dropin.NavigationView (com.mapbox.navigation.qa_test_app.view.NavigationViewFragment received Fragment#onDestroyView() callback (references to its views should be cleared to prevent leaks)) with key 92d7d59f-0ad1-4300-821d-f9af3bc5a4ce
2022-05-19 16:14:03.305 11947-11947/com.mapbox.navigation.qa_test_app D/LeakCanary: Watching instance of com.mapbox.navigation.qa_test_app.view.NavigationViewFragment (com.mapbox.navigation.qa_test_app.view.NavigationViewFragment received Fragment#onDestroy() callback) with key a64665db-7008-45ce-bf9a-ee2bbcc67429
2022-05-19 16:14:03.476 11947-11991/com.mapbox.navigation.qa_test_app D/LeakCanary: Setting up flushing for Thread[MapboxRenderThread,5,main]
2022-05-19 16:15:25.471 11947-11947/com.mapbox.navigation.qa_test_app D/LeakCanary: Watching instance of androidx.lifecycle.ReportFragment (androidx.lifecycle.ReportFragment received Fragment#onDestroy() callback) with key b14a06a7-47af-4be6-bd17-7d9429af7f16
2022-05-19 16:15:25.471 11947-11947/com.mapbox.navigation.qa_test_app D/LeakCanary: Watching instance of com.mapbox.navigation.qa_test_app.view.MapboxNavigationViewFragmentActivity (com.mapbox.navigation.qa_test_app.view.MapboxNavigationViewFragmentActivity received Activity#onDestroy() callback) with key e4d36217-8446-47bd-802a-1646675d28d2
2022-05-19 16:15:25.538 11947-11947/com.mapbox.navigation.qa_test_app D/LeakCanary: Watching instance of com.mapbox.navigation.dropin.NavigationView (com.mapbox.navigation.qa_test_app.view.NavigationViewFragment received Fragment#onDestroyView() callback (references to its views should be cleared to prevent leaks)) with key a062514e-fc52-444b-be41-e15409c5a008
2022-05-19 16:15:25.538 11947-11947/com.mapbox.navigation.qa_test_app D/LeakCanary: Watching instance of com.mapbox.navigation.qa_test_app.view.NavigationViewFragment (com.mapbox.navigation.qa_test_app.view.NavigationViewFragment received Fragment#onDestroy() callback) with key dddd1d90-8768-4f2c-bbae-247c6039975d
```

**Comment by kmadsen** [comment]
Here is an issue with the camera (also being surfaced by this change but not actually produced by this change)

1. Start active navigation in **portrait** mode
2. Notice the camera state goes to FOLLOWING
3. Press the button to go to ROUTE_OVERVIEW
4. Change orientation to **landscape** mode
5. Notice the camera state is FOLLOWING

Expecting it to be in ROUTE_OVERVIEW since that was the last selection

**Review by kmadsen — APPROVED** [review]
Only minor nitpicks and issues that are out of scope of this change.

It looks great!

**Pull request merged**